### PR TITLE
Protocol 12.40

### PR DESCRIPTION
**Pull request by FakeShinoda**
- Only work with clients 12.40
- Client 12.40.10030 support

### CODE REVIEW

**Comment by MarceloM3** [comment]
a

**Comment by majestyotbr** [comment]
Tested, working. Thanks for your contribution!

**Comment by majestyotbr** [comment]
Client 12.40.10030 available for download in:
https://forums.otserv.com.br/index.php?/forums/topic/167933-otservbr-global-cliente-tibia-12/

**Review by majestyotbr — APPROVED** [review]

File: src/protocolgame.cpp [hunk 1]
```diff
@@ -268,13 +268,9 @@ void ProtocolGame::onRecvFirstMessage(NetworkMessage& msg)
...
 	msg.skipBytes(3); // U16 dat revision, game preview state
 
-	if (clientVersion >= 1149 && clientVersion < 1200) {
-		// on 1149.6xxx, this was removed later.
-		// extra byte for "optimise connection stability"
-		if (msg.getLength() - msg.getBufferPosition() > 128) {
-			shouldAddExivaRestrictions = true;
-			msg.skipBytes(1);
-		}
+	// In version 12.40.10030 we have 13 extra bytes
```
Comment (jo3bingham) [inline src/protocolgame.cpp:271]: It's not just 13 extra bytes, [it's a string](https://github.com/jo3bingham/TibiaAPI/commit/f71c0b76ef5019cb35fa0248f9eed0e021b9dae8) containing the client version.

**Pull request merged**